### ISSUE TITLE
When assigning a group color to a group, not all entries are marked

### ISSUE THREAD

**Issue by koppor** [JabRef/jabref#7325]
1. Open library with existing group from old JabRef version
2. Select group
3. Edit group color  
  ![grafik](https://user-images.githubusercontent.com/1366654/104136662-dd209800-5397-11eb-9efb-07791c215838.png)
4. Dialog "Change of Grouping Method" appears.  
  ![grafik](https://user-images.githubusercontent.com/1366654/104136670-e9a4f080-5397-11eb-9559-33a4cda12fde.png)
5. Click "OK"
6. Group is unselected --> maintable changes
7. Click on group
8. Correct number of entries is displayed, group color not shown
  ![grafik](https://user-images.githubusercontent.com/1366654/104136703-0d683680-5398-11eb-859a-1b0bc52c46d9.png)

In another experiment, only 1/3 of the entries got the group color

Questions:

- Why is the dialog "Change of Grouping Method" appearing?
- Why are not all (or none) entries marked with a color?

**Comment by koppor** [comment]
```diff
@@ -22797,11 +22797,11 @@ it is less complex than WS-R and well-integrated in the WSPA.},
 1 KeywordGroup:XACT\;0\;keywords\;XACT\;0\;0\;1\;\;\;\;;
 1 KeywordGroup:BPEL\;0\;keywords\;BPEL\;0\;0\;1\;\;\;\;;
 1 KeywordGroup:prio1\;0\;keywords\;prio1\;1\;0\;1\;\;\;\;;
 1 KeywordGroup:ZEUS\;0\;keywords\;ZEUS\;1\;0\;1\;\;\;\;;
 1 KeywordGroup:SOC\;0\;keywords\;SOC\;0\;0\;1\;\;\;\;;
-1 KeywordGroup:OG\;0\;keywords\;OG\;0\;0\;0\;\;\;\;;
+1 KeywordGroup:OG\;0\;keywords\;OG\;0\;0\;1\;0xccffccff\;\;\;;
 1 KeywordGroup:AskedForPDF\;0\;keywords\;AskedForPDF\;1\;0\;1\;\;\;\;;
 1 KeywordGroup:FetchPDF\;0\;keywords\;FetchPDF\;1\;0\;1\;\;\;\;;
 1 KeywordGroup:Rigorosum\;0\;keywords\;Rigorosum\;0\;0\;1\;\;\;\;;
 1 KeywordGroup:ChorNotion\;0\;keywords\;ChorNotion\;0\;0\;1\;\;\;\;;
 1 SearchGroup:Meine paper\;0\;author=kopp and author=oliver\;0\;0\;1\;\;\;\;;
```

**Comment by koppor** [comment]
@k3KAW8Pnf7mkmdSMPHz27 May I ask whether you could have a look here?

**Comment by k3KAW8Pnf7mkmdSMPHz27** [comment]
> * Why is the dialog "Change of Grouping Method" appearing?

I have never noticed the **name** of the dialog window before.

My assumption, so far, is that it appears because when you make changes to a group it often implies changes in the bibentries as well. E.g., an explicit group will modify the GROUP field of all entries if you change its name, and the same should happen for all Keyword groups. It is perhaps a little bit misleading and appear in too many cases. I could probably look at this at some point this week. I'd guess the required changes are small.

> * Why are not all (or none) entries marked with a color?

There are two issues,
1. The color region is only updated on changes in the GROUP field (and this group uses the KEYWORDS field). See https://github.com/JabRef/jabref/pull/7173#pullrequestreview-565525360 , and feel free to chime in.
2. The other part is that you are creating a change in the actual group (change of color), which should be a meta data event. It is addressed for `TexGroup`s in #7173. It is similar to point 1 in that I can see an argument for making it both in and out of scope of #7173 . I'll add another comment in that PR when I have time. I am not sure how to minimize the performance impact of this (which is why I only addressed it for `TexGroup`s)

**Comment by k3KAW8Pnf7mkmdSMPHz27** [comment]
(Just to state it plainly, I'll look into it after #7173 is merged, it doesn't make sense to do so before)

**Comment by k3KAW8Pnf7mkmdSMPHz27** [comment]
This is still open and I am not currently working on it. It ended up spilling needing further down the line changes and I forgot to write about that here,
